src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,43 @@
 'use strict';
 
 class Animal {
-  // write your code here
-}
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 
+  static delleteDeadAnimals() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
+}
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    if (
+      target instanceof Carnivore ||
+      (target instanceof Herbivore && target.hidden)
+    ) {
+      return `${this.name} can't bite ${target.name}`;
+    } else {
+      target.health -= 50;
+      Animal.delleteDeadAnimals();
```

Comment:
Consider renaming the method `delleteDeadAnimals` to `deleteDeadAnimals` after fixing the typo, and ensure that all references to this method are updated accordingly.